File: 프로그래머스/lv1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> student1 = {1, 2, 3, 4, 5};//5
    vector<int> student2 = {2, 1, 2, 3, 2, 4, 2, 5};//8
    vector<int> student3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};//10
    int cnt[3] = {0, };
    for(int i = 0; i < answers.size(); i++){
        if(answers[i] == student1[i % 5]) cnt[0]++;
        if(answers[i] == student2[i % 8]) cnt[1]++;
       	if(answers[i] == student3[i % 10]) cnt[2]++;
    }
    int max = *max_element(cnt, cnt+3);
    for(int i = 0; i < 3; i++){
        if(max == cnt[i]) answer.push_back(i + 1);
    }

    sort(answer.begin(), answer.end());
    return answer;
}

File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        
        int[] first = {1,2,3,4,5};
        int[] second = {2,1,2,3,2,4,2,5};
        int[] third = {3,3,1,1,2,2,4,4,5,5};
        
        int[] count = {0,0,0};
        
        // 점수 계산
        for(int i=0; i<answers.length; i++){
            if(first[i%first.length] == answers[i]) count[0]++;
            if(second[i%second.length] == answers[i]) count[1]++;
            if(third[i%third.length] == answers[i]) count[2]++;
        }
        
        // 최대 점수 구하기
        int max = Math.max(count[0], Math.max(count[1], count[2]));
        
        // 최대 점수를 가진 수포자 리스트에 저장
        List<Integer> list = new ArrayList<>();
        for(int i=0; i<count.length; i++) if(max==count[i]) list.add(i+1);
        
        int[] answer = new int[list.size()];
        for(int i=0; i<answer.length; i++) answer[i] = list.get(i);
        return answer;
    }
}